prime-router/src/main/kotlin/common/HttpClientUtils.kt

```diff
@@ -530,44 +558,96 @@ class HttpClientUtils {
          * note: most configuration parameters are overridable
          * e.g. expectSuccess default to false because most of the time
          * the caller wants to handle the whole range of response status
-         * @param bearerTokens null default, the access token needed to call the endpoint
+         * @param bearerTokens the access token needed to call the endpoint
          * @return a HttpClient with all sensible defaults
          */
-        fun createDefaultHttpClient(accessToken: String?): HttpClient {
-            return HttpClient(Apache) {
-                // installs logging into the call to post to the server
-                // commented out - not to override underlying default logger settings
-                // enable to trace http client internals when needed
-                // install(Logging) {
-                //     logger = Logger.SIMPLE
-                //     level = LogLevel.INFO
-                // }
-                // not using Bearer Auth handler due to refresh token behavior
-                accessToken?.let {
-                    defaultRequest {
-                        header("Authorization", "Bearer $it")
+        fun getDefaultHttpClient(accessToken: String?): HttpClient {
+            synchronized(this) {
+                if (accessToken != null) {
+                    return getDefaultHttpClientWithAuth(accessToken)
+                } else {
+                    httpClient ?: HttpClient(Apache) {
+                        install(ContentNegotiation) {
+                                json(
+                                    Json {
+                                        prettyPrint = true
+                                        isLenient = true
+                                        ignoreUnknownKeys = true
+                                    }
+                                )
+                            }
+                        install(HttpTimeout)
+                        engine {
+                                followRedirects = true
+                                socketTimeout = TIMEOUT
+                                connectTimeout = TIMEOUT
+                                connectionRequestTimeout = TIMEOUT
+                                customizeClient {
+                                }
```

Comment:
I did not look closely at it. Do you want it gone? I'm not wed to it being there. I carried it over as you said

Reply:
It would be nice to clean up while we're in here.

I did a cursory search through [docs](https://ktor.io/docs/client-engines.html#java) and I don't think an empty `customizeClient {}` block does anything. So we should be able to remove it without any impact.

Or if something does break (ideally not in prod 😅) we can add a comment there.

Reply:
I'll take care of it